crates/sui-types/src/messages.rs

```diff
@@ -81,10 +81,12 @@ pub enum CallArg {
 pub enum ObjectArg {
     // A Move object, either immutable, or owned mutable.
     ImmOrOwnedObject(ObjectRef),
-    // A Move object that's shared and mutable.
+    // A Move object that's shared.
+    // SharedObject::mutable controls whether caller asks for a mutable reference to shared object.
     SharedObject {
         id: ObjectID,
         initial_shared_version: SequenceNumber,
+        mutable: bool,
```

Comment:
I'd suggest using an enum rather than a bool param, so that if we need a third option in the future the protocol evolution is simpler.

(No, I can't think of what the third option would be either)

Reply:
I think we should be able to update type of the field without making it a enum right away (otherwise do we need to make any `bool` in the system into enum?)